crates/rome_cli/src/commands/format.rs

```
@@ -149,7 +123,16 @@ pub(crate) fn format(mut session: CliSession) {
         }
 
         let name = path.display().to_string();
-        let source = fs::read_to_string(path).ok().unwrap_or_else(String::new);
+        let mut source = String::new();
+        session
+            .app
+            .fs
+            .open(&path)
+            .and_then(|mut file| file.read_to_string(&mut source))
+            // Because the diagnostics include IO errors, path could be a
+            // directory or a non-existent file so any error that may happen
+            // while reading the file is ignored
```

I don't think I understand this comment. Why is it safe to ignore the error?

---

There are a few separate things going here and the comment maybe doesn't explain it too well:
- The first is that if this code is reached this means an error diagnostic was emitted for this path, we can't really know what it was at this stage since its an opaque diagnostic but it could be that the file doesn't exist, is a directory, or the process doesn't have the permission to read it. There's a fairly high chance the same error will happen again when the file is loaded a second time, in which case we don't want to do anything with it since we're already in the process of printing the error diagnostic to the console anyway and we don't want to show the same error twice
- The second scenario is that the filesystem could be in an inconsistent state, for instance the file got deleted between the moment the diagnostic was emitted and the moment it gets printed. The probability of this happening is very low so for now I just opted to show the diagnostics in "degraded mode" with no code span information and not print any additional error. 

One thing I want to change in the future is how filesystem abstractions are composed, so that it would be possible to wrap an `OsFileSystem` inside a `MemoryFileSystem` instead of using either one of the other: the idea is that the file would only be loaded once from the OS then kept into memory, and all subsequent reads would be performed on that byte buffer instead which would ensure a coherent view of the filesystem for the duration of the CLI run as well as avoiding the file reloading that currently happens for diagnostics.

---

I think the comment should be reworded. Your explanation is way better compared to the comment. If in the future I would have to read it again, it would have hard time to understand it without the context given by this explanation of yours.